crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/execution/entry_point.rs

```diff
@@ -172,18 +174,23 @@ pub fn execute_call_entry_point(
     });
 
     // region: Modified blockifier code
-    match &result {
-        Ok((call_info, syscall_counter)) => remove_syscall_resources_and_exit_success_call(
-            call_info,
-            syscall_counter,
-            context,
-            resources,
-            runtime_state,
-        ),
-        Err(err) => exit_error_call(err, runtime_state, resources, entry_point),
+    match result {
+        Ok((call_info, syscall_counter, vm_trace)) => {
```

Comment:
If the trace is heavy, won't this affect our memory footprint/perf significantly? 
Have you checked this?

Reply:
I think this is worth benchmarking

Reply:
This line or this PR in general? The PR prob will. I can collect the traces conditionally (only when `--save-trace-data` or `--build-profile` flag is used), WDYT?

Reply:
That's reasonable for me. Although i would check the performance anyway.

Reply:
They changed nothing for our current benchmarks, it prob will for larger tests, but my approach will be the same anyways

cc @cptartur - first benchmark usage 🎉